Dead code elimination issue "use of undeclared type `B`" in slint compiled rust

I've created [a branch where you can reproduce this](https://github.com/colelawrence/slint/tree/bug/2023-04/compiler-issue-with-pruning-types) from within the slint repo.

### Setup
```rs
// main.rs
slint::include_modules!();

pub fn main() {
    let main_window = MainWindow::new().unwrap();
    main_window.run().unwrap();
}
```
```rs
// build.rs
fn main() {
    slint_build::compile("./main_window.slint").unwrap();
}
```

```qml
// main_window.slint
import { A, B, Create } from "./types.slint";
// must be separate file

component ShowB {
    in property <B> b: Create.b(1099);
}

component ShowA {
    property <A> a: Create.a(1001);
    Text { text: "A \{root.a.id}"; } // required
}

export component MainWindow inherits Window {
    ShowA {}
}
```

```qml
// types.slint
export struct A { id: int }
export struct B { id: int }
export global Create {
    function a(id: int) -> A {
        return { id: id };
    }
    function b(id: int) -> B {
        return { id: id };
    }
}
```

### Error
```sh
   Compiling issue_repro v1.0.3 (/Users/cole/dev/slint/examples/issue_repro)
warning: Component is neither used nor exported
warning:  --> /Users/cole/dev/slint/examples/issue_repro/./main_window.slint:4:1
warning:   |
warning: 4 | component ShowB {
warning:   | ^
error[E0412]: cannot find type `B` in this scope
  --> /Users/cole/dev/slint/target/debug/build/issue_repro-a9c2e463849a06c9/out/main_window.rs:92:116
   |
3  |      # [derive (Default , PartialEq , Debug , Clone)] pub struct r#A {
   |                                                       -------------- similarly named struct `A` defined here
...
92 |          # [allow (dead_code , unused)] pub fn r#fn_b (self : :: core :: pin :: Pin < & Self > , arg_0 : i32 ,) -> r#B {
   |                                                                                                                    ^^^ help: a struct with a similar name exists: `A`

error[E0433]: failed to resolve: use of undeclared type `B`
  --> /Users/cole/dev/slint/target/debug/build/issue_repro-a9c2e463849a06c9/out/main_window.rs:98:43
   |
98 |                      let mut the_struct = r#B :: default () ;
   |                                           ^^^
   |                                           |
   |                                           use of undeclared type `B`
   |                                           help: a struct with a similar name exists: `A`

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
The following warnings were emitted during compilation:

warning: Component is neither used nor exported
warning:  --> /Users/cole/dev/slint/examples/issue_repro/./main_window.slint:4:1
warning:   |
warning: 4 | component ShowB {
warning:   | ^

error: could not compile `issue_repro` due to 2 previous errors
```

<details>
<summary>main_window.rs</summary>

```rust
# [allow (non_snake_case)] # [allow (non_camel_case_types)] # [allow (clippy :: style)] # [allow (clippy :: complexity)] # [allow (unused_braces)] # [allow (clippy :: erasing_op)] # [allow (clippy :: approx_constant)] # [allow (clippy :: eq_op)] # [allow (clippy :: cmp_owned)] # [allow (clippy :: redundant_clone)] mod slint_generatedMainWindow {
     use slint :: private_unstable_api :: re_exports :: * ;
     # [derive (Default , PartialEq , Debug , Clone)] pub struct r#A {
         pub r#id : i32 }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin] struct InnerColorSchemeSelector_5 {
         r#dark_color_scheme : slint :: private_unstable_api :: re_exports :: Property < bool > , root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , }
     impl InnerColorSchemeSelector_5 {
         fn new () -> :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> {
             slint :: private_unstable_api :: re_exports :: Rc :: pin (Self :: default ()) }
         fn init (self : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) {
             # ! [allow (unused)] self . root . set (VRc :: downgrade (root)) ;
             let self_rc = self ;
             let _self = self_rc . as_ref () ;
             {
                 slint :: private_unstable_api :: set_property_binding ({
                     * & InnerColorSchemeSelector_5 :: FIELD_OFFSETS . r#dark_color_scheme }
                 . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_ref () ;
                     (_self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap () . dark_color_scheme ()) as _ }
                ) ;
                 }
             }
         }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin] struct InnerPalette_6 {
         r#neutralDark : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: Color > , r#white : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: Color > , root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , }
     impl InnerPalette_6 {
         fn new () -> :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> {
             slint :: private_unstable_api :: re_exports :: Rc :: pin (Self :: default ()) }
         fn init (self : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) {
             # ! [allow (unused)] self . root . set (VRc :: downgrade (root)) ;
             let self_rc = self ;
             let _self = self_rc . as_ref () ;
             {
                 slint :: private_unstable_api :: set_property_binding ({
                     * & InnerPalette_6 :: FIELD_OFFSETS . r#neutralDark }
                 . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_ref () ;
                     (if ! InnerColorSchemeSelector_5 :: FIELD_OFFSETS . r#dark_color_scheme . apply_pin (_self . root . get () . unwrap () . upgrade () . unwrap () . globals . global_ColorSchemeSelector_5 . as_ref ()) . get () {
                         slint :: private_unstable_api :: re_exports :: Color :: from_argb_encoded (4280295198f64 as u32) }
                     else {
                         (slint :: private_unstable_api :: re_exports :: Color :: from_argb_encoded (4294243572f64 as u32)) as _ }
                    ) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding ({
                     * & InnerPalette_6 :: FIELD_OFFSETS . r#white }
                 . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_ref () ;
                     (if ! InnerColorSchemeSelector_5 :: FIELD_OFFSETS . r#dark_color_scheme . apply_pin (_self . root . get () . unwrap () . upgrade () . unwrap () . globals . global_ColorSchemeSelector_5 . as_ref ()) . get () {
                         slint :: private_unstable_api :: re_exports :: Color :: from_argb_encoded (4294967295f64 as u32) }
                     else {
                         (slint :: private_unstable_api :: re_exports :: Color :: from_argb_encoded (4280229663f64 as u32)) as _ }
                    ) as _ }
                ) ;
                 }
             }
         }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin] struct InnerStyleMetrics_7 {
         root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , }
     impl InnerStyleMetrics_7 {
         fn new () -> :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> {
             slint :: private_unstable_api :: re_exports :: Rc :: pin (Self :: default ()) }
         fn init (self : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) {
             # ! [allow (unused)] self . root . set (VRc :: downgrade (root)) ;
             let self_rc = self ;
             let _self = self_rc . as_ref () ;
             }
         }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin] struct InnerCreate_8 {
         root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , }
     impl InnerCreate_8 {
         fn new () -> :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> {
             slint :: private_unstable_api :: re_exports :: Rc :: pin (Self :: default ()) }
         fn init (self : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < Self >> , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) {
             # ! [allow (unused)] self . root . set (VRc :: downgrade (root)) ;
             let self_rc = self ;
             let _self = self_rc . as_ref () ;
             }
         # [allow (dead_code , unused)] pub fn r#fn_a (self : :: core :: pin :: Pin < & Self > , arg_0 : i32 ,) -> r#A {
             let _self = self ;
             let args = (arg_0 ,) ;
             ({
                 return ({
                     let obj = (args . 0 . clone () as i32 ,) ;
                     let mut the_struct = r#A :: default () ;
                     the_struct . r#id = obj . 0 as _ ;
                     the_struct }
                ) as _ ;
                 }
            ) }
         # [allow (dead_code , unused)] pub fn r#fn_b (self : :: core :: pin :: Pin < & Self > , arg_0 : i32 ,) -> r#B {
             let _self = self ;
             let args = (arg_0 ,) ;
             ({
                 return ({
                     let obj = (args . 0 . clone () as i32 ,) ;
                     let mut the_struct = r#B :: default () ;
                     the_struct . r#id = obj . 0 as _ ;
                     the_struct }
                ) as _ ;
                 }
            ) }
         }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin] struct InnerShowA_root_3 {
         r#root_3 : slint :: private_unstable_api :: re_exports :: r#Empty , r#text_4 : slint :: private_unstable_api :: re_exports :: r#Text , r#root_3_a : slint :: private_unstable_api :: re_exports :: Property < r#A > , r#root_3_layoutinfo_h : slint :: private_unstable_api :: re_exports :: Property < r#LayoutInfo > , r#root_3_layoutinfo_v : slint :: private_unstable_api :: re_exports :: Property < r#LayoutInfo > , r#root_3_text_4_min_height : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: LogicalLength > , r#root_3_text_4_min_width : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: LogicalLength > , r#root_3_text_4_preferred_height : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: LogicalLength > , r#root_3_text_4_preferred_width : slint :: private_unstable_api :: re_exports :: Property < slint :: private_unstable_api :: re_exports :: LogicalLength > , self_weak : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeakMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerShowA_root_3 >> , window_adapter : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: Rc < dyn slint :: private_unstable_api :: re_exports :: WindowAdapter >> , root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , tree_index : :: core :: cell :: Cell < u32 > , tree_index_of_first_child : :: core :: cell :: Cell < u32 > , }
     impl InnerShowA_root_3 {
         pub fn init (self_rc : slint :: private_unstable_api :: re_exports :: VRcMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , Self > , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow > , tree_index : u32 , tree_index_of_first_child : u32) {
             # ! [allow (unused)] # ! [allow (unused)] let _self = self_rc . as_pin_ref () ;
             _self . self_weak . set (VRcMapped :: downgrade (& self_rc)) ;
             _self . root . set (VRc :: downgrade (root)) ;
             _self . window_adapter . set (root . window_adapter . get () . unwrap () . clone ()) ;
             _self . tree_index . set (tree_index) ;
             _self . tree_index_of_first_child . set (tree_index_of_first_child) ;
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_a }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (_self . root . get () . unwrap () . upgrade () . unwrap () . globals . global_Create_8 . as_ref () . r#fn_a (1001f64 as _)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_h }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     ((({
                         let mut the_struct = r#LayoutInfo :: default () ;
                         the_struct . r#max = 340282346638528860000000000000000000000f64 as _ ;
                         the_struct . r#max_percent = 100f64 as _ ;
                         the_struct . r#min = 0f64 as _ ;
                         the_struct . r#min_percent = 0f64 as _ ;
                         the_struct . r#preferred = 0f64 as _ ;
                         the_struct . r#stretch = 1f64 as _ ;
                         the_struct }
                    ) + (slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())))) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_v }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     ((({
                         let mut the_struct = r#LayoutInfo :: default () ;
                         the_struct . r#max = 340282346638528860000000000000000000000f64 as _ ;
                         the_struct . r#max_percent = 100f64 as _ ;
                         the_struct . r#min = 0f64 as _ ;
                         the_struct . r#min_percent = 0f64 as _ ;
                         the_struct . r#preferred = 0f64 as _ ;
                         the_struct . r#stretch = 1f64 as _ ;
                         the_struct }
                    ) + (slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())))) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_min_height }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ((slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())) . r#min as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_min_width }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ((slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())) . r#min as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_preferred_height }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ((slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())) . r#preferred as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_preferred_width }
                ) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ((slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                    ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ())) . r#preferred as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#color) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: Brush :: SolidColor (InnerPalette_6 :: FIELD_OFFSETS . r#neutralDark . apply_pin (_self . root . get () . unwrap () . upgrade () . unwrap () . globals . global_Palette_6 . as_ref ()) . get ())) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#height) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ({
                         let r#minmax_lhs39 = ({
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_preferred_height }
                        ) . apply_pin (_self) . get () . get () ;
                         ;
                         let r#minmax_rhs39 = ({
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_min_height }
                        ) . apply_pin (_self) . get () . get () ;
                         ;
                         if ((r#minmax_lhs39 . clone () as f64) > (r#minmax_rhs39 . clone () as f64)) {
                             r#minmax_lhs39 . clone () }
                         else {
                             (r#minmax_rhs39 . clone ()) as _ }
                         }
                     as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#text) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (((((slint :: private_unstable_api :: re_exports :: SharedString :: from ("A ")) + (slint :: private_unstable_api :: re_exports :: SharedString :: from (slint :: private_unstable_api :: re_exports :: format ! ("{}" , (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_a }
                    ) . apply_pin (_self) . get ()) . r#id) . as_str ()) . as_str ()))) + (slint :: private_unstable_api :: re_exports :: SharedString :: from ("") . as_str ()))) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#width) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ({
                         let r#minmax_lhs38 = ({
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_preferred_width }
                        ) . apply_pin (_self) . get () . get () ;
                         ;
                         let r#minmax_rhs38 = ({
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_text_4_min_width }
                        ) . apply_pin (_self) . get () . get () ;
                         ;
                         if ((r#minmax_lhs38 . clone () as f64) > (r#minmax_rhs38 . clone () as f64)) {
                             r#minmax_lhs38 . clone () }
                         else {
                             (r#minmax_rhs38 . clone ()) as _ }
                         }
                     as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#x) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new (((((({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                     + r#Empty :: FIELD_OFFSETS . r#width) . apply_pin (_self) . get () . get () as f64) - (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                     + r#Text :: FIELD_OFFSETS . r#width) . apply_pin (_self) . get () . get () as f64)) as f64) / (2f64 as f64)) as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#y) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new (((((({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                     + r#Empty :: FIELD_OFFSETS . r#height) . apply_pin (_self) . get () . get () as f64) - (({
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                     + r#Text :: FIELD_OFFSETS . r#height) . apply_pin (_self) . get () . get () as f64)) as f64) / (2f64 as f64)) as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#font_family) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#font_size) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#font_weight) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#horizontal_alignment) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#letter_spacing) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#overflow) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#vertical_alignment) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
             + r#Text :: FIELD_OFFSETS . r#wrap) . apply_pin (_self) . set_constant () ;
             }
         pub fn user_init (self_rc : slint :: private_unstable_api :: re_exports :: VRcMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , Self >) {
             let _self = self_rc . as_pin_ref () ;
             }
         fn visit_dynamic_children (self : :: core :: pin :: Pin < & Self > , dyn_index : usize , order : slint :: private_unstable_api :: re_exports :: TraversalOrder , visitor : slint :: private_unstable_api :: re_exports :: ItemVisitorRefMut) -> slint :: private_unstable_api :: re_exports :: VisitChildrenResult {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             }
         fn layout_info (self : :: core :: pin :: Pin < & Self > , orientation : slint :: private_unstable_api :: re_exports :: Orientation) -> slint :: private_unstable_api :: re_exports :: LayoutInfo {
             # ! [allow (unused)] let _self = self ;
             match orientation {
                 slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal => ({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_h }
                ) . apply_pin (_self) . get () , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical => ({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_v }
                ) . apply_pin (_self) . get () , }
             }
         fn subtree_range (self : :: core :: pin :: Pin < & Self > , dyn_index : usize) -> slint :: private_unstable_api :: re_exports :: IndexRange {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             }
         fn subtree_component (self : :: core :: pin :: Pin < & Self > , dyn_index : usize , subtree_index : usize , result : & mut slint :: private_unstable_api :: re_exports :: ComponentWeak) {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             ;
             }
         fn index_property (self : :: core :: pin :: Pin < & Self >) -> usize {
             # ! [allow (unused)] let _self = self ;
             core :: usize :: MAX }
         fn accessible_role (self : :: core :: pin :: Pin < & Self > , index : usize) -> slint :: private_unstable_api :: re_exports :: AccessibleRole {
             # ! [allow (unused)] let _self = self ;
             match index {
                 1usize => slint :: private_unstable_api :: re_exports :: r#AccessibleRole :: r#Text , _ => AccessibleRole :: default () , }
             }
         fn accessible_string_property (self : :: core :: pin :: Pin < & Self > , index : usize , what : slint :: private_unstable_api :: re_exports :: AccessibleStringProperty ,) -> slint :: private_unstable_api :: re_exports :: SharedString {
             # ! [allow (unused)] let _self = self ;
             match (index , what) {
                 (1usize , AccessibleStringProperty :: r#Label) => ({
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
                 + r#Text :: FIELD_OFFSETS . r#text) . apply_pin (_self) . get () , _ => Default :: default () , }
             }
         }
     # [derive (slint :: private_unstable_api :: re_exports :: FieldOffsets , Default)] # [const_field_offset (slint :: private_unstable_api :: re_exports :: const_field_offset)] # [repr (C)] # [pin_drop] pub struct InnerMainWindow {
         r#root_1 : slint :: private_unstable_api :: re_exports :: r#WindowItem , r#showa_2 : InnerShowA_root_3 , self_weak : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeakMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , window_adapter : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: Rc < dyn slint :: private_unstable_api :: re_exports :: WindowAdapter >> , root : slint :: private_unstable_api :: re_exports :: OnceCell < slint :: private_unstable_api :: re_exports :: VWeak < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >> , tree_index : :: core :: cell :: Cell < u32 > , tree_index_of_first_child : :: core :: cell :: Cell < u32 > , globals : Globals_MainWindow }
     impl InnerMainWindow {
         pub fn init (self_rc : slint :: private_unstable_api :: re_exports :: VRcMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , Self > , root : & slint :: private_unstable_api :: re_exports :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow > , tree_index : u32 , tree_index_of_first_child : u32) {
             # ! [allow (unused)] # ! [allow (unused)] let _self = self_rc . as_pin_ref () ;
             _self . self_weak . set (VRcMapped :: downgrade (& self_rc)) ;
             _self . root . set (VRc :: downgrade (root)) ;
             _self . window_adapter . set (root . window_adapter . get () . unwrap () . clone ()) ;
             _self . tree_index . set (tree_index) ;
             _self . tree_index_of_first_child . set (tree_index_of_first_child) ;
             InnerShowA_root_3 :: init (VRcMapped :: map (self_rc . clone () , | x | {
                 * & Self :: FIELD_OFFSETS . r#showa_2 }
             . apply_pin (x)) , & _self . root . get () . unwrap () . upgrade () . unwrap () , tree_index_of_first_child + 1u32 - 1 , tree_index_of_first_child + 2u32 - 1) ;
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
                 + r#WindowItem :: FIELD_OFFSETS . r#background) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: Brush :: SolidColor (InnerPalette_6 :: FIELD_OFFSETS . r#white . apply_pin (_self . root . get () . unwrap () . upgrade () . unwrap () . globals . global_Palette_6 . as_ref ()) . get ())) as _ }
                ) ;
                 }
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#title) . apply_pin (_self) . set ({
                 (slint :: private_unstable_api :: re_exports :: SharedString :: from ("Slint Window")) as slint :: private_unstable_api :: re_exports :: SharedString }
            ) ;
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                 + {
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                 + r#Empty :: FIELD_OFFSETS . r#height) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ({
                         let r#minmax_lhs41 = (({
                             InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                         + {
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_v }
                        ) . apply_pin (_self) . get ()) . r#preferred ;
                         ;
                         let r#minmax_rhs41 = (({
                             InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                         + {
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_v }
                        ) . apply_pin (_self) . get ()) . r#min ;
                         ;
                         if ((r#minmax_lhs41 . clone () as f64) > (r#minmax_rhs41 . clone () as f64)) {
                             r#minmax_lhs41 . clone () }
                         else {
                             (r#minmax_rhs41 . clone ()) as _ }
                         }
                     as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                 + {
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                 + r#Empty :: FIELD_OFFSETS . r#width) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new ({
                         let r#minmax_lhs40 = (({
                             InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                         + {
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_h }
                        ) . apply_pin (_self) . get ()) . r#preferred ;
                         ;
                         let r#minmax_rhs40 = (({
                             InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                         + {
                             * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3_layoutinfo_h }
                        ) . apply_pin (_self) . get ()) . r#min ;
                         ;
                         if ((r#minmax_lhs40 . clone () as f64) > (r#minmax_rhs40 . clone () as f64)) {
                             r#minmax_lhs40 . clone () }
                         else {
                             (r#minmax_rhs40 . clone ()) as _ }
                         }
                     as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                 + {
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                 + r#Empty :: FIELD_OFFSETS . r#x) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new (((((({
                         * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
                     + r#WindowItem :: FIELD_OFFSETS . r#width) . apply_pin (_self) . get () . get () as f64) - (({
                         InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                     + {
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                     + r#Empty :: FIELD_OFFSETS . r#width) . apply_pin (_self) . get () . get () as f64)) as f64) / (2f64 as f64)) as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             {
                 slint :: private_unstable_api :: set_property_binding (({
                     InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                 + {
                     * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                 + r#Empty :: FIELD_OFFSETS . r#y) . apply_pin (_self) , & self_rc , move | self_rc | {
                     let _self = self_rc . as_pin_ref () ;
                     (slint :: private_unstable_api :: re_exports :: LogicalLength :: new (((((({
                         * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
                     + r#WindowItem :: FIELD_OFFSETS . r#height) . apply_pin (_self) . get () . get () as f64) - (({
                         InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
                     + {
                         * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
                     + r#Empty :: FIELD_OFFSETS . r#height) . apply_pin (_self) . get () . get () as f64)) as f64) / (2f64 as f64)) as slint :: private_unstable_api :: re_exports :: Coord)) as _ }
                ) ;
                 }
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#always_on_top) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#default_font_family) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#default_font_weight) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#icon) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#no_frame) . apply_pin (_self) . set_constant () ;
             ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
             + r#WindowItem :: FIELD_OFFSETS . r#title) . apply_pin (_self) . set_constant () ;
             }
         pub fn user_init (self_rc : slint :: private_unstable_api :: re_exports :: VRcMapped < slint :: private_unstable_api :: re_exports :: ComponentVTable , Self >) {
             let _self = self_rc . as_pin_ref () ;
             InnerShowA_root_3 :: user_init (VRcMapped :: map (self_rc . clone () , | x | {
                 * & Self :: FIELD_OFFSETS . r#showa_2 }
             . apply_pin (x)) ,) ;
             }
         fn visit_dynamic_children (self : :: core :: pin :: Pin < & Self > , dyn_index : usize , order : slint :: private_unstable_api :: re_exports :: TraversalOrder , visitor : slint :: private_unstable_api :: re_exports :: ItemVisitorRefMut) -> slint :: private_unstable_api :: re_exports :: VisitChildrenResult {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             }
         fn layout_info (self : :: core :: pin :: Pin < & Self > , orientation : slint :: private_unstable_api :: re_exports :: Orientation) -> slint :: private_unstable_api :: re_exports :: LayoutInfo {
             # ! [allow (unused)] let _self = self ;
             match orientation {
                 slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal => slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                     * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
                ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Horizontal , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ()) , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical => slint :: private_unstable_api :: re_exports :: Item :: layout_info (({
                     * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
                ) . apply_pin (_self) , slint :: private_unstable_api :: re_exports :: Orientation :: Vertical , _self . root . get () . unwrap () . upgrade () . unwrap () . window_adapter . get () . unwrap ()) , }
             }
         fn subtree_range (self : :: core :: pin :: Pin < & Self > , dyn_index : usize) -> slint :: private_unstable_api :: re_exports :: IndexRange {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             }
         fn subtree_component (self : :: core :: pin :: Pin < & Self > , dyn_index : usize , subtree_index : usize , result : & mut slint :: private_unstable_api :: re_exports :: ComponentWeak) {
             # ! [allow (unused)] let _self = self ;
             match dyn_index {
                 _ => panic ! ("invalid dyn_index {}" , dyn_index) , }
             ;
             }
         fn index_property (self : :: core :: pin :: Pin < & Self >) -> usize {
             # ! [allow (unused)] let _self = self ;
             core :: usize :: MAX }
         fn accessible_role (self : :: core :: pin :: Pin < & Self > , index : usize) -> slint :: private_unstable_api :: re_exports :: AccessibleRole {
             # ! [allow (unused)] let _self = self ;
             match index {
                 1usize => {
                     * & Self :: FIELD_OFFSETS . r#showa_2 }
                 . apply_pin (_self) . accessible_role (0) , 2usize ..= 2usize => {
                     * & Self :: FIELD_OFFSETS . r#showa_2 }
                 . apply_pin (_self) . accessible_role (index - 2usize + 1) , _ => AccessibleRole :: default () , }
             }
         fn accessible_string_property (self : :: core :: pin :: Pin < & Self > , index : usize , what : slint :: private_unstable_api :: re_exports :: AccessibleStringProperty ,) -> slint :: private_unstable_api :: re_exports :: SharedString {
             # ! [allow (unused)] let _self = self ;
             match (index , what) {
                 (1usize , _) => {
                     * & Self :: FIELD_OFFSETS . r#showa_2 }
                 . apply_pin (_self) . accessible_string_property (0 , what) , (2usize ..= 2usize , _) => {
                     * & Self :: FIELD_OFFSETS . r#showa_2 }
                 . apply_pin (_self) . accessible_string_property (index - 2usize + 1 , what) , _ => Default :: default () , }
             }
         }
     impl InnerMainWindow {
         pub fn new () -> core :: result :: Result < vtable :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , Self > , slint :: PlatformError , > {
             # ! [allow (unused)] let window_adapter = slint :: private_unstable_api :: create_window_adapter () ? ;
             let mut _self = Self :: default () ;
             let self_rc = VRc :: new (_self) ;
             let _self = self_rc . as_pin_ref () ;
             _self . window_adapter . set (window_adapter) ;
             slint :: private_unstable_api :: re_exports :: WindowInner :: from_pub (_self . window_adapter . get () . unwrap () . window ()) . set_component (& VRc :: into_dyn (self_rc . clone ())) ;
             slint :: private_unstable_api :: re_exports :: register_component (_self , Self :: item_array () , & self_rc . window_adapter . get () . unwrap ()) ;
             Self :: init (slint :: private_unstable_api :: re_exports :: VRc :: map (self_rc . clone () , | x | x) , & self_rc , 0 , 1) ;
             core :: result :: Result :: Ok (self_rc) }
         fn item_tree () -> & 'static [slint :: private_unstable_api :: re_exports :: ItemTreeNode] {
             const ITEM_TREE : [slint :: private_unstable_api :: re_exports :: ItemTreeNode ;
             3usize] = [slint :: private_unstable_api :: re_exports :: ItemTreeNode :: Item {
                 is_accessible : false , children_count : 1u32 , children_index : 1u32 , parent_index : 0u32 , item_array_index : 0u32 , }
             , slint :: private_unstable_api :: re_exports :: ItemTreeNode :: Item {
                 is_accessible : false , children_count : 1u32 , children_index : 2u32 , parent_index : 0u32 , item_array_index : 1u32 , }
             , slint :: private_unstable_api :: re_exports :: ItemTreeNode :: Item {
                 is_accessible : true , children_count : 0u32 , children_index : 3u32 , parent_index : 1u32 , item_array_index : 2u32 , }
            ] ;
             & ITEM_TREE }
         fn item_array () -> & 'static [vtable :: VOffset < Self , ItemVTable , vtable :: AllowPin >] {
             static ITEM_ARRAY : slint :: private_unstable_api :: re_exports :: OnceBox < [vtable :: VOffset < InnerMainWindow , ItemVTable , vtable :: AllowPin > ;
             3usize] > = slint :: private_unstable_api :: re_exports :: OnceBox :: new () ;
             & * ITEM_ARRAY . get_or_init (|| Box :: new ([VOffset :: new ({
                 * & InnerMainWindow :: FIELD_OFFSETS . r#root_1 }
            ) , VOffset :: new ({
                 InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
             + {
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#root_3 }
            ) , VOffset :: new ({
                 InnerMainWindow :: FIELD_OFFSETS . r#showa_2 }
             + {
                 * & InnerShowA_root_3 :: FIELD_OFFSETS . r#text_4 }
            )])) }
         }
     impl slint :: private_unstable_api :: re_exports :: PinnedDrop for InnerMainWindow {
         fn drop (self : core :: pin :: Pin < & mut InnerMainWindow >) {
             use slint :: private_unstable_api :: re_exports :: * ;
             ComponentVTable_static ! (static VT for self :: InnerMainWindow) ;
             new_vref ! (let vref : VRef < ComponentVTable > for Component = self . as_ref () . get_ref ()) ;
             slint :: private_unstable_api :: re_exports :: unregister_component (self . as_ref () , vref , Self :: item_array () , self . window_adapter . get () . unwrap ()) ;
             }
         }
     impl slint :: private_unstable_api :: re_exports :: Component for InnerMainWindow {
         fn visit_children_item (self : :: core :: pin :: Pin < & Self > , index : isize , order : slint :: private_unstable_api :: re_exports :: TraversalOrder , visitor : slint :: private_unstable_api :: re_exports :: ItemVisitorRefMut) -> slint :: private_unstable_api :: re_exports :: VisitChildrenResult {
             return slint :: private_unstable_api :: re_exports :: visit_item_tree (self , & VRcMapped :: origin (& self . as_ref () . self_weak . get () . unwrap () . upgrade () . unwrap ()) , self . get_item_tree () . as_slice () , index , order , visitor , visit_dynamic) ;
             # [allow (unused)] fn visit_dynamic (_self : :: core :: pin :: Pin < & InnerMainWindow > , order : slint :: private_unstable_api :: re_exports :: TraversalOrder , visitor : ItemVisitorRefMut , dyn_index : usize) -> VisitChildrenResult {
                 _self . visit_dynamic_children (dyn_index , order , visitor) }
             }
         fn get_item_ref (self : :: core :: pin :: Pin < & Self > , index : usize) -> :: core :: pin :: Pin < ItemRef > {
             match & self . get_item_tree () . as_slice () [index] {
                 ItemTreeNode :: Item {
                     item_array_index , .. }
                 => {
                     Self :: item_array () [* item_array_index as usize] . apply_pin (self) }
                 ItemTreeNode :: DynamicTree {
                     .. }
                 => panic ! ("get_item_ref called on dynamic tree") , }
             }
         fn get_item_tree (self : :: core :: pin :: Pin < & Self >) -> slint :: private_unstable_api :: re_exports :: Slice < slint :: private_unstable_api :: re_exports :: ItemTreeNode > {
             Self :: item_tree () . into () }
         fn get_subtree_range (self : :: core :: pin :: Pin < & Self > , index : usize) -> slint :: private_unstable_api :: re_exports :: IndexRange {
             self . subtree_range (index) }
         fn get_subtree_component (self : :: core :: pin :: Pin < & Self > , index : usize , subtree_index : usize , result : & mut slint :: private_unstable_api :: re_exports :: ComponentWeak) {
             self . subtree_component (index , subtree_index , result) ;
             }
         fn subtree_index (self : :: core :: pin :: Pin < & Self >) -> usize {
             self . index_property () }
         fn parent_node (self : :: core :: pin :: Pin < & Self > , _result : & mut slint :: private_unstable_api :: re_exports :: ItemWeak) {
             }
         fn layout_info (self : :: core :: pin :: Pin < & Self > , orientation : slint :: private_unstable_api :: re_exports :: Orientation) -> slint :: private_unstable_api :: re_exports :: LayoutInfo {
             self . layout_info (orientation) }
         fn accessible_role (self : :: core :: pin :: Pin < & Self > , index : usize) -> slint :: private_unstable_api :: re_exports :: AccessibleRole {
             self . accessible_role (index) }
         fn accessible_string_property (self : :: core :: pin :: Pin < & Self > , index : usize , what : slint :: private_unstable_api :: re_exports :: AccessibleStringProperty , result : & mut slint :: private_unstable_api :: re_exports :: SharedString ,) {
             * result = self . accessible_string_property (index , what) ;
             }
         }
     pub struct r#MainWindow (vtable :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) ;
     impl r#MainWindow {
         pub fn new () -> core :: result :: Result < Self , slint :: PlatformError > {
             let inner = InnerMainWindow :: new () ? ;
             inner . globals . global_ColorSchemeSelector_5 . clone () . init (& inner) ;
             inner . globals . global_Palette_6 . clone () . init (& inner) ;
             inner . globals . global_StyleMetrics_7 . clone () . init (& inner) ;
             inner . globals . global_Create_8 . clone () . init (& inner) ;
             InnerMainWindow :: user_init (slint :: private_unstable_api :: re_exports :: VRc :: map (inner . clone () , | x | x)) ;
             core :: result :: Result :: Ok (Self (inner)) }
         }
     impl From < r#MainWindow > for vtable :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow > {
         fn from (value : r#MainWindow) -> Self {
             value . 0 }
         }
     impl slint :: ComponentHandle for r#MainWindow {
         type Inner = InnerMainWindow ;
         fn as_weak (& self) -> slint :: Weak < Self > {
             slint :: Weak :: new (& self . 0) }
         fn clone_strong (& self) -> Self {
             Self (self . 0 . clone ()) }
         fn from_inner (inner : vtable :: VRc < slint :: private_unstable_api :: re_exports :: ComponentVTable , InnerMainWindow >) -> Self {
             Self (inner) }
         fn run (& self) -> core :: result :: Result < () , slint :: PlatformError > {
             self . show () ? ;
             slint :: run_event_loop () ? ;
             self . hide () ? ;
             core :: result :: Result :: Ok (()) }
         fn show (& self) -> core :: result :: Result < () , slint :: PlatformError > {
             self . window () . show () }
         fn hide (& self) -> core :: result :: Result < () , slint :: PlatformError > {
             self . window () . hide () }
         fn window (& self) -> & slint :: Window {
             vtable :: VRc :: as_pin_ref (& self . 0) . get_ref () . window_adapter . get () . unwrap () . window () }
         fn global < 'a , T : slint :: Global < 'a , Self >> (& 'a self) -> T {
             T :: get (& self) }
         }
     # [allow (dead_code)] struct Globals_MainWindow {
         global_ColorSchemeSelector_5 : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < InnerColorSchemeSelector_5 >> , global_Palette_6 : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < InnerPalette_6 >> , global_StyleMetrics_7 : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < InnerStyleMetrics_7 >> , global_Create_8 : :: core :: pin :: Pin < slint :: private_unstable_api :: re_exports :: Rc < InnerCreate_8 >> , }
     impl Default for Globals_MainWindow {
         fn default () -> Self {
             Self {
                 global_ColorSchemeSelector_5 : InnerColorSchemeSelector_5 :: new () , global_Palette_6 : InnerPalette_6 :: new () , global_StyleMetrics_7 : InnerStyleMetrics_7 :: new () , global_Create_8 : InnerCreate_8 :: new () , }
             }
         }
     const _THE_SAME_VERSION_MUST_BE_USED_FOR_THE_COMPILER_AND_THE_RUNTIME : slint :: VersionCheck_1_0_3 = slint :: VersionCheck_1_0_3 ;
     }
 pub use slint_generatedMainWindow :: {
     r#MainWindow , r#A }
 ;
 pub use slint :: {
     ComponentHandle as _ , Global as _ , ModelExt as _ }
 ;
```
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,10 @@ All notable changes to this project are documented in this file.
 
 ## Unreleased
 
+### General
+
+ * Fixed missing items compilation error in the generated code related to public functions (#2655).
+
 ### Rust
 
 - Added `slint::Image::load_from_svg_data(buffer: &[u8])` to load SVGs from memory.

--- a/cspell.json
+++ b/cspell.json
@@ -93,6 +93,7 @@
         "SPDX",
         "streetsidesoftware",
         "struct",
+        "structs",
         "tabwidget",
         "testcase",
         "textedit",

--- a/internal/compiler/passes/collect_structs.rs
+++ b/internal/compiler/passes/collect_structs.rs
@@ -94,6 +94,12 @@ fn visit_named_object(ty: &Type, visitor: &mut impl FnMut(&String, &Type)) {
                 visit_named_object(a, visitor);
             }
         }
+        Type::Function { return_type, args } => {
+            visit_named_object(return_type, visitor);
+            for a in args {
+                visit_named_object(a, visitor);
+            }
+        }
         _ => {}
     }
 }

--- a/tests/cases/types/functions.slint
+++ b/tests/cases/types/functions.slint
@@ -1,0 +1,32 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+// Test that structs work if they are only referenced by functions
+
+struct Foo1 { member: int, }
+struct Foo2 { a: Foo1 }
+struct Foo3 { b: int }
+
+TestCase := TouchArea {
+    // Based on Issue #2655
+    public function fn1(foo2: Foo2) -> Foo3 {
+      return { b: foo2.a.member+1 };
+    }
+}
+
+
+/*
+```rust
+let instance = TestCase::new().unwrap();
+
+assert_eq!(instance.invoke_fn1(Foo2{ a: Foo1{ member: 123 } }).b, 124);
+```
+
+```cpp
+auto handle = TestCase::create();
+const TestCase &instance = *handle;
+
+assert_eq(instance.invoke_fn1(Foo2{ Foo1{ 123 } }).b, 124);
+```
+
+*/

